truffleruby/src/main/java/org/truffleruby/core/CoreLibrary.java

```diff
@@ -1099,6 +1099,10 @@ public static boolean fitsIntoInteger(long value) {
         return ((int) value) == value;
     }
 
+    public static boolean fitsIntoUnsignedInteger(long value) {
+        return value == (value & 0xffffffffl) || value < 0 && value >= Integer.MIN_VALUE;
```

Comment:
I think it's best to keep our internals clean. If we don't need it at some caller, then call a normal implementation. Only use this extended range when needed.
And still, why these obscure conditions over `Integer.MIN_VALUE <= value && value < (1L << 64)`?